kafka-connect-s3/src/main/java/io/confluent/connect/s3/S3SinkConnectorConfig.java

```diff
@@ -613,6 +616,19 @@ public String toString() {
     }
   }
 
+  private static class BucketNameValidator implements ConfigDef.Validator {
+    @Override
+    public void ensureValid(String name, Object bucket) {
+      String bucketName = ((String) bucket).trim();
+      try {
+        BucketNameUtils.validateBucketName(bucketName);
+      } catch (IllegalArgumentException e) {
+        throw new ConfigException(e.getMessage() + ". " + bucketName
+                + " is an invalid bucket name. Does not follow AWS guidelines.");
```

Comment:
Might be better to swap the order and put the most important part first:
```suggestion
        throw new ConfigException(
            String.format(
                "'%s' is not a valid bucket name and does not follow AWS guidelines. %s",
                bucketName,
                e.getMessage()
            )
        );
```